tests/tests/test-author-mapping-keys.ts

```diff
@@ -0,0 +1,325 @@
+const faith = keyring.addFromUri(FAITH_PRIVATE_KEY, null, "ethereum");
+
+describeDevMoonbeam("Author Mapping - Set Faith first time keys", (context) => {
+  before("setup account & keys", async function () {
+    debug(`Setting account ${faith.address} keys: ${concatOriginalKeys}`);
+    await context.polkadotApi.tx.authorMapping.setKeys(concatOriginalKeys).signAndSend(faith);
+    await context.createBlock();
+  });
+
+  it("should succeed", async function () {
+    const { extrinsic, resultEvent } = await getBlockExtrinsic(
+      context.polkadotApi,
+      await context.polkadotApi.rpc.chain.getBlockHash(),
+      "authorMapping",
+      "setKeys"
+    );
+
+    expect(extrinsic).to.exist;
+    expect(resultEvent.method).to.equal("ExtrinsicSuccess");
+  });
+
+  it("should send AuthorRegistered event", async function () {
+    const { events } = await getBlockExtrinsic(
+      context.polkadotApi,
+      await context.polkadotApi.rpc.chain.getBlockHash(),
+      "authorMapping",
+      "setKeys"
+    );
+    expect(events.find((e) => e.section == "authorMapping" && e.method == "AuthorRegistered")).to
+      .exist;
+  });
+
+  it("should set new keys", async function () {
+    const mapping = (await context.polkadotApi.query.authorMapping.mappingWithDeposit(
+      originalKeys[0]
+    )) as any;
+    expect(mapping.isSome).to.be.true;
+    expect(mapping.unwrap().account.toString()).to.equal(faith.address);
+    expect(mapping.unwrap().keys_.toString()).to.equal(originalKeys[1]);
+  });
+
+  it("should set correct nimbus lookup", async function () {
+    const nimbusLookup = (await context.polkadotApi.query.authorMapping.nimbusLookup(
+      faith.address
+    )) as any;
+    expect(nimbusLookup.isSome).to.be.true;
+    expect(nimbusLookup.unwrap().toString()).to.equal(originalKeys[0]);
+  });
+});
+
+describeDevMoonbeam("Author Mapping - Update Faith mapping to the same keys", (context) => {
+  before("setup account & keys", async function () {
+    await context.polkadotApi.tx.authorMapping.setKeys(concatOriginalKeys).signAndSend(faith);
+    await context.createBlock();
+
+    // Updating with the same keys
+    await context.polkadotApi.tx.authorMapping.setKeys(concatOriginalKeys).signAndSend(faith);
+    await context.createBlock();
+  });
+
+  it("should succeed", async function () {
+    const { extrinsic, resultEvent } = await getBlockExtrinsic(
+      context.polkadotApi,
+      await context.polkadotApi.rpc.chain.getBlockHash(),
+      "authorMapping",
+      "setKeys"
+    );
+
+    expect(extrinsic).to.exist;
+    expect(resultEvent.method).to.equal("ExtrinsicSuccess");
+  });
+
+  it("should send AuthorRotated event", async function () {
+    const { events } = await getBlockExtrinsic(
+      context.polkadotApi,
+      await context.polkadotApi.rpc.chain.getBlockHash(),
+      "authorMapping",
+      "setKeys"
+    );
+    expect(events.find((e) => e.section == "authorMapping" && e.method == "AuthorRotated")).to
```

Comment:
This event is sent even when the keys are the same. Should we do that or only when the keys are actually changing? 
cc: @4meta5

Reply:
> This event is sent even when the keys are the same. Should we do that or only when the keys are actually changing?

I think the event should be emitted whenever the extrinsic is successfully called. The event structure is
```rust
KeysRotated {
	new_nimbus_id: NimbusId,
	account_id: T::AccountId,
	new_keys: T::Keys,
},
```
Users can use these fields to check if the `new_nimbus_id` equals the old one or `new_keys` equals the old keys.